lib/ecto/adapters/sql.ex

```diff
@@ -282,7 +289,13 @@ defmodule Ecto.Adapters.SQL do
       """
     end
 
-    {pool_name, pool_opts} = repo.__pool__
+    # Check if the pool options should overriden
+    {pool_name, pool_opts} = case Keyword.fetch(opts, :pool) do
+      {:ok, pool} when pool != Ecto.Adapters.SQL.Sandbox ->
```

Comment:
This was the reason the build was broken initially. I didn't want to add any new options like `should_override_pool: true` and instead chose to guard against `Ecto.Adapters.SQL.Sandbox` explicitly.